RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -128,242 +160,742 @@ void L2MuonSeedGeneratorFromL1T::produce(edm::Event& iEvent, const edm::EventSet
+                          TrajectorySeed::const_iterator
+                          tsci  = selOffseeds[theL1][theOffs]->recHits().first,
+                          tscie = selOffseeds[theL1][theOffs]->recHits().second;
+                          for(; tsci!=tscie; ++tsci) {
+                            newContainer.push_back(*tsci);
+                          }
+                          output->push_back(L2MuonTrajectorySeed(seedTSOS,newContainer,alongMomentum,
+                                            MuonRef(muColl,  distance(muColl->begin(muColl->getFirstBX()),it)  )));
+          }
+        }
+      }
+
+      else if(matchType==3) {
+
+        for(nL1=0; nL1 < nMuColl; ++nL1) {
+          double tempDR = 999;
+          unsigned int theL1 = 0;
+          unsigned int theOffs = 0;
+          auto theit = muColl->begin(muColl->getFirstBX());
+
+          // L1Q > 10 (L1Q = 12)
+          for(i=0; i<nMuColl; ++i) {
+            theit = muColl->begin(muColl->getFirstBX()) + i;
+            if (theit->hwQual() > 10) {
+              for(j=0; j<nOfflineSeed1; ++j) {
+                if(tempDR > dRmtx[i][j]) {
+                  tempDR = dRmtx[i][j];
+                  theL1 = i;
+                  theOffs = j;
+                }
+              }
+            }
+          }
+
+          // 6 < L1Q <= 10 (L1Q = 8)
+          if (tempDR == 999) {
+            for(i=0; i<nMuColl; ++i) {
+              theit = muColl->begin(muColl->getFirstBX()) + i;
+              if ( (theit->hwQual() <= 10) && (theit->hwQual() > 6) ) {
+                for(j=0; j<nOfflineSeed1; ++j) {
+                  if(tempDR > dRmtx[i][j]) {
+                    tempDR = dRmtx[i][j];
+                    theL1 = i;
+                    theOffs = j;
+                  }
+                }
+              }
+            }
+          }
+
+          // L1Q <= 6 (L1Q = 4)
+          if (tempDR == 999) {
+            for(i=0; i<nMuColl; ++i) {
+              theit = muColl->begin(muColl->getFirstBX()) + i;
+              if (theit->hwQual() <= 6) {
+                for(j=0; j<nOfflineSeed1; ++j) {
+                  if(tempDR > dRmtx[i][j]) {
+                    tempDR = dRmtx[i][j];
+                    theL1 = i;
+                    theOffs = j;
+                  }
+                }
+              }
+            }
+          }
+
+          auto it = muColl->begin(muColl->getFirstBX()) + theL1;
+
+          double newDRcone = matchingDR[0];
+          if ( fabs(it->eta()) < etaBins.back() ){
+            std::vector<double>::iterator lowEdge = std::upper_bound (etaBins.begin(), etaBins.end(), fabs(it->eta()));
+            newDRcone    =  matchingDR.at( lowEdge - etaBins.begin() - 1);
+          }
+
+          if( !(tempDR < newDRcone) )  continue;
+
+          // Remove row and column for given (L1mu, offSeed)
+          for(i=0; i<nMuColl; ++i) {
+            dRmtx[i][theOffs] = 999;
+          }
+          for(j=0; j<nOfflineSeed1; ++j) {
```

Comment:
The same here: some gain could be obtained if you define some possible ```bool removed_column[OfflineSeed1]``` and check it in the if clauses at L767, L782, L798 before comparing the two "dR"